### PR TITLE
Support generic type of download file

### LINKED ISSUE #293: Support user download changes in webviz-core-components
**Is your feature request related to a problem? Please describe.**
Allow the plugin consumers to change the name of the file that is downloaded to the client macing when the download button is clicked. Also, allow for other types of files to be downloaded i.e a simple CSV file without it needing to be part of an archive. 

The solution should provide support for the suggested implementation in https://github.com/equinor/webviz-core-components/issues/71 
related to issue https://github.com/equinor/webviz-core-components/issues/70

### PR DESCRIPTION
**Pull request by DanSava**
*Insert a description of your pull request (PR) here, and check off the boxes below when they are done.*

---

### Contributor checklist

- [x] :tada: This PR closes #293 .
- [ ] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [x] :book: I have considered adding a new entry in `CHANGELOG.md`, and added it if should be communicated there.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -156,10 +156,11 @@ callback is set. A typical data download callback will look like
 @app.callback(self.plugin_data_output,
               [self.plugin_data_requested])
```
Comment (anders-kiaer) [inline CONTRIBUTING.md:157]:
```suggestion
@app.callback(self.plugin_data_output,
              self.plugin_data_requested)
```
The recent Dash versions (`>=1.15`) now supports input not being a list, so we can use this opportunity to simplify the guide probably. We already require `>=1.16` in `webviz-config` due to CSP and usage of https://github.com/plotly/dash/pull/1371.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -170,8 +171,8 @@ The attributes `self.plugin_data_output` and `self.plugin_data_requested`
 are Dash `Output` and `Input` instances respectively, and are provided by
 the base class `WebvizPluginABC` (i.e. include them as shown here).
 
-The function `WebvizPluginABC.plugin_data_compress` is a utility function
-which takes a list of dictionaries, giving filenames and corresponding data,
+The function `WebvizPluginABC.plugin_compressed_data` is a utility function
+which takes a file name and a list of dictionaries, containing file names and corresponding data,
 and compresses them to a zip archive which is then downloaded by the user.
```
Comment (anders-kiaer) [inline CONTRIBUTING.md:176]: Would be nice to add an example here of calculating `base64` and add mime-type directly by the plugin, e.g. of a CSV (which is a typical use case).

**Review by anders-kiaer — COMMENTED** [review]
Looks good @DanSava 🎉 Some small suggestions to consider. 

If you are busy the next days I will probably have some time next week to get this to the finish line if you want 🥅.

**Comment by DanSava** [comment]
Thank you for the review and suggestions I will try to update the PR today I also need to update the `webviz-core-components` PR to account for the `mime_type` which I forgot to add.

**Review by HansKallekleiv — APPROVED** [review]
LGTM 👍

**Pull request merged**